scylla/src/transport/session.rs

```diff
@@ -732,6 +743,58 @@ impl Session {
 
         return Err(last_error);
     }
+
+    pub async fn await_schema_agreement(&self) -> Result<(), QueryError> {
+        while !self.check_schema_agreement().await? {
+            tokio::time::sleep(self.schema_agreement_interval).await
+        }
+        Ok(())
+    }
+
+    pub async fn await_timed_schema_agreement(
+        &self,
+        timeout_duration: u64,
+    ) -> Result<bool, QueryError> {
+        if timeout(
+            Duration::from_millis(timeout_duration),
+            self.await_schema_agreement(),
+        )
+        .await
+        .is_err()
+        {
+            return Ok(false);
+        }
+        Ok(true)
+    }
+
+    async fn check_schema_agreement(&self) -> Result<bool, QueryError> {
+        let peers_rows = self.query(PEERS_VERSION, &[]);
+        let local_version = self.fetch_schema_version().await?;
```

Comment:
Those two queries should be executed on the same connection, otherwise if you query `system.local` on node `X` but query `system.peers` on node `Y`, then you will completely miss the schema version of node `Y`. Now, the load balancing policy will most likely choose two different nodes for both queries.

Reply:
You can also send both queries in parallel in order to decrease the latency of the operation - for example by using the [`tokio::try_join!`](https://docs.rs/tokio/1.5.0/tokio/macro.try_join.html) macro.